apps/drupal/src/Plugin/Preprocessor/PreprocessorManager.php

```diff
@@ -0,0 +1,509 @@
+<?php
```

Comment:
This seems like a lot of code for it to need to exist per-Drupal-theme. Should it be moved to a contrib module which Particle depends on (I can see why you'd hate having to do that, but just for the sake of discussion I wanted to ask it) or at least moved out of the individual app so that if there are multiple Drupal themes they can all reference the same file instead of each having copies?

Or is this file meant to be owned/altered by the project, in which case it makes sense for it to be here? But that would open up a whole new discussion - that this is a lot of pretty deep code to ask each project to understand and maintain.

Reply:
To me it makes most sense for the theme and it's tools to own their implementation. But I don't disagree in that this file is a bit much. I think this ought to be broken up and tuned. But definitely wanted v1 to be a fairly direct port, and then open up to us at large to tune it.